docs/src/catalyst_applications/ode_simulation_performance.md

```diff
@@ -0,0 +1,318 @@
+
+## [Regarding stiff and non-stiff problems and solvers](@id ode_simulation_performance_stiffness)
+Generally, ODE problems can be categorised into [*stiff ODEs* and *non-stiff ODEs*](https://en.wikipedia.org/wiki/Stiff_equation). This categorisation is important due to stiff ODEs requiring specialised solvers. A common cause of failure to simulate an ODE is the use of a non-stiff solver for a stiff problem. There is no exact way to determine whether a given ODE is stiff or not, however, systems with several different time scales (e.g. a CRN with both slow and fast reactions) typically generate stiff ODEs.
+
+Here we simulate the (stiff) [brusselator](https://en.wikipedia.org/wiki/Brusselator) model using the `Tsit5` solver (which is designed for non-stiff ODEs):
+```@example ode_simulation_performance_1
+using Catalyst, OrdinaryDiffEq, Plots
+
+brusselator = @reaction_network begin
+    A, ∅ --> X
+    1, 2X + Y --> 3X
+    B, X --> Y
+    1, X --> ∅
+end
+
+u0 = [:X => 1.0, :Y => 0.0]
+tspan = (0.0, 20.0)
+p = [:A => 10.0, :B => 40.0]
+oprob = ODEProblem(brusselator, u0, tspan, p)
+
+sol1 = solve(oprob, Tsit5())
+plot(sol1)
+``` 
+We note that we get a warning, indicating that an instability was detected (the typical indication of a non-stiff solver being used for a stiff ODE). Furthermore, the resulting plot ends at $t≈10$, meaning that the simulation was not completed. Indeed, we can confirm this by checking the *return code* of the solution object:
+```@example ode_simulation_performance_1
+sol1.retcode
+```
+Next, we instead try the `Rodas5P` solver (which is designed for stiff problems):
+```@example ode_simulation_performance_1
+sol2 = solve(oprob, Rodas5P())
+plot(sol2)
+``` 
+This time the simulation was successfully completed, which can be confirmed by checking the return code:
+```@example ode_simulation_performance_1
+sol2.retcode
+``` 
+
+Generally, ODE solvers can be divided into [*explicit* and *implicit* solvers](https://en.wikipedia.org/wiki/Explicit_and_implicit_methods). Roughly, explicit solvers (which include `Tsit5`) are better for non-stiff problems, with implicit solvers (like `Rodas5P`) being required for stiff problems. While we could use implicit solvers for all problems (to guarantee successful simulations irrespective of stiffness), these are typically less performant on non-stiff problems (as compared to the explicit solvers). An important property of implicit solvers is that they require the *computation of a Jacobian* as part of their routine. This means that the various options for efficient Jacobian computation [described later in this tutorial](@ref ode_simulation_performance_jacobian) are only relevant to implicit solvers.
+
+Finally, we should note that this is not an exact science, and sometimes explicit solvers can successfully solve a stiff problem. E.g. if we change the parameters values of our previous Brusselator model to `[:A => 1.0, :B => 4.0]`, the `Tsit5` will successfully be able to simulate it.
```

Comment:
This is not correct. Changing the parameters can change a problem from stiff to non-stiff. It is not only about equation form.

Reply:
If there is a fundamental principle about stiffness, it's based on the pseudospectra of the Jacobian. A reasonable simplification is the condition number of the Jacobian, i.e. the ratio between the maximum and minimum eigenvalue, or roughly the scale between the fastest and slowest operations. So it's not the case that parameters don't effect that. And in direct contrast, it's is almost certainly a function of parameters.

Reply:
fixed